isamples_sesar/SesarTransformer.py

```diff
@@ -0,0 +1,630 @@
+    #     return None
+
+    # Disabled pending resolution of https://github.com/isamplesorg/isamples_inabox/issues/255
+    def has_material_categories(self) -> typing.List[str]:
+        material = self._material_type()
+        # TODO: implement predictions
+        # if not material:
+        #     prediction_results = self._compute_material_prediction_results()
+        #     if prediction_results is not None:
+        #         return [prediction.value for prediction in prediction_results]
+        #     else:
+        #         return []
+        return MaterialCategoryMetaMapper.categories(material)
+
+    def has_specimen_categories(self) -> typing.List[str]:
+        sample_type = self.sample.sample_type.name
+        return SpecimenCategoryMetaMapper.categories(sample_type)
+
+    def id_string(self) -> str:
+        return "https://data.isamples.org/digitalsample/{0}/{1}".format(
+            'igsn', self.sample.igsn
+        )
+
+    def _material_type(self) -> str:
+        if self.sample.classification and self.sample.top_level_classification:
+            return f"{self.sample.classification.name}>{self.sample.top_level_classification.name}"
+        elif self.sample.classification:
+            return self.sample.classification.name
+        elif self.sample.top_level_classification:
+            return self.sample.top_level_classification.name
+        return ""
+
+    @staticmethod
+    def _logger():
+        return logging.getLogger("isamples_metadata.SESARTransformer")
+
+    def sample_label(self) -> str:
+        return self.sample.name
+
+    def sample_identifier_string(self) -> str:
+        return f"IGSN:{self.sample.igsn}"
+
+    def sample_description(self) -> str:
+        description_arr = []
+
+        classification = self._material_type()
+        if classification:
+            description_arr.append(classification)
+
+        if self.sample.collection_method:
+            description_arr.append(self.sample.collection_method)
+
+        if self.sample.description:
+            description_arr.append(self.sample.description)
+
+        if description_arr:
+            description = ". ".join(description_arr)
+            return description
+        return Transformer.NOT_PROVIDED
+
+    def informal_classification(self) -> typing.List[str]:
+        """Not currently used for SESAR"""
+        return [Transformer.NOT_PROVIDED]
+
+    def keywords(self) -> typing.List:
+        keyword_arr = []
+        sample_type = self.sample.sample_type
+        if sample_type:
+            parent_sample_type = self.sample.sample_type.parent_sample_type
+            if parent_sample_type:
+                sample_type_str = f"{parent_sample_type.name}>{sample_type.name}"
+            else:
+                sample_type_str = sample_type.name
+            keyword_arr.append({
+                "keyword": sample_type_str,
+                "scheme_name": "SESAR: Sample Type"
```

Comment:
Generally speaking I think these schemes are supposed to be URIs.  I'll double check what OpenContext is doing.

Reply:
this is just copying what i saw in the examples